test/functional/testlib.bash

```diff
@@ -1031,6 +1122,108 @@ destroy_test_environment() {
 
 }
 
+# creates a web server to host fake swupd content with or without certifiates
+start_web_server() {
+
+	local port
+	local server_args
+	local server_pid
+	local status
+
+	cb_usage() {
+		echo $(cat <<-EOF
+		Usage:
+		    start_web_server [-k] <server priv key> [-p] <server pub key> [-s]
+
+		Options:
+		    start_web_server [-c] <client pub key> [-p] <server pub key> [-k] <server priv key> [-s]
+
+		Options:
+		    -c    Path to public key to be used for client certificate authentication
+		    -k    Path to server private key which must correspond to the provided server public key
+		    -p    Path to server public key which enables SSL authentication
+		    -s    Use a slow update server
+
+		Notes:
+		    - When the server is using SSL authentication, a pair of corresponding public and private keys must be passed
+		      as arguments.
+
+		Example of usage:
+
+		    The following command will create a web server that uses SSL authentication and inserts a delay for content
+		    updates.
+
+		    start_web_server -k /private_key.pem -p /public_key.pem -s
+
+		EOF
+		)
+	}
+
+	set -f  # turn off globbing
+	while getopts :c:k:p:s opt; do
+		case "$opt" in
+			c)	server_args="$server_args --client_cert $OPTARG" ;;
+			k)	server_args="$server_args --server_key $OPTARG" ;;
+			p)	server_args="$server_args --server_cert $OPTARG" ;;
+			s)	server_args="$server_args --slow_server" ;;
+			*)	cb_usage
+				return ;;
+		esac
+	done
+	set +f  # turn globbing back on
+
+	# start web server and write port/pid numbers to their respective files
+	python3 $FUNC_DIR/server.py $server_args --port_file $PORT_FILE &
```

Comment:
I will clean these up throughout this pull request, but I think I have one exception. The server_args variable should be split on spaces so that argparse can properly read each argument.